lib/server/services/run/RunService.js

```diff
@@ -135,11 +137,23 @@ class RunService {
             ? await getOrCreateAllDetectorsByName(newRun.detectors.split(',').map((value) => value.trim()))
             : null;
 
+        let runType = null;
         if (runTypeName) {
-            const runType = await getOrCreateRunType({ name: runTypeName });
+            runType = await getOrCreateRunType({ name: runTypeName });
             newRun.runTypeId = runType.id;
         }
 
+        let lhcFill = null;
+        if (newRun.fillNumber) {
+            lhcFill = await getLhcFill(newRun.fillNumber);
+        }
+
+        newRun.definition = getRunDefinition({
```

Comment:
I do not think that `createRun` is the correct place to calculate the runDefinition because AliECS on this step does not send yet the `timeO2Start` and `timeTrgStart` which means that the `getRunDefinition` will not be able to identify an overlap between the run and the stable beam.

This information comes only at a latter stage in `updateRun`, right?

Reply:
As discussed together, it's in the low level services:
In the LHC fill update and in the run creation and update